file_writer_control/WorkerCommandChannel.py

```diff
@@ -0,0 +1,95 @@
+from file_writer_control.CommandHandler import CommandHandler
+from file_writer_control.WorkerFinder import WorkerFinder
+from file_writer_control.WriteJob import WriteJob
+from file_writer_control.WorkerStatus import WorkerState, WorkerStatus
+from file_writer_control.JobStatus import JobState
+from random import randrange
+import threading
+import time
+from typing import List
+import atexit
+
+START_JOB_TIMEOUT = 30  # Seconds
+SEND_JOB_TIMEOUT = 10  # Seconds
+
+
+class WorkerCommandChannel(WorkerFinder):
+    """
+    A child of WorkerFinder intended for use with "direct command" style of starting a file-writing job.
+    """
+    def __init__(self, command_topic_url: str):
+        super().__init__(command_topic_url)
+        self.start_job_threads = []
+        self._stop_event = threading.Event()
+
+        def do_exit():
+            self.stop_threads()
+        atexit.register(do_exit)
+
+    def stop_threads(self):
+        self._stop_event.set()
+        for thread in self.start_job_threads:
+            thread.join()
+
+    def try_start_job(self, job: WriteJob) -> CommandHandler:
+        """
+        Attempt to start a file-writing job by finding an available (idle) worker and sending it a write command.
+        Will re-try to send a write command until a timeout has been reached. This member function is not blocking.
+        :param job: The job to be started.
+        :return: A CommandHandler for (more) easily checking the outcome of the start command.
+        """
+        thread_kwargs = {"do_job": job, "stop_event": self._stop_event}
+        self.command_channel.add_job_id(job.job_id)
+        self.command_channel.add_command_id(job.job_id, job.job_id)
```

Comment:
As far as I see from the base code, `add_command_id` takes a `job_id` and a `command_id`. In this case are they identical and equal to `job_id`?

Reply:
Ah, yes. For this particular case (start job commands) this is correct. Because this the first use of that job-id, it is also re-used as a command-id.